rust/lance-datagen/src/generator.rs

```diff
@@ -1663,11 +1717,7 @@ pub mod array {
                 TimeUnit::Microsecond => rand::<DurationMicrosecondType>(),
                 TimeUnit::Nanosecond => rand::<DurationNanosecondType>(),
             },
-            DataType::Interval(unit) => match unit {
-                IntervalUnit::DayTime => rand::<IntervalDayTimeType>(),
-                IntervalUnit::MonthDayNano => rand::<IntervalMonthDayNanoType>(),
-                IntervalUnit::YearMonth => rand::<IntervalYearMonthType>(),
-            },
+            DataType::Interval(unit) => rand_interval(*unit),
```

Comment:
Why the change here?

Reply:
I think they changed the underlying native type to a struct rather than i128/i64. Thus `Distribution` was no longer implemented.

https://github.com/apache/arrow-rs/pull/5769